### PR TITLE
Jormungandr: add ridesharing conf

### PR DESCRIPTION
**Pull request by xlqian**
```json
{
 ...  
  "ridesharing": [{
    "class": "jormungandr.scenarios.ridesharing.instant_system.InstantSystem",
    "args" : {
        "service_url" : "url",
        "api_key" : "api_key" ,
        "network": "Rennes",
        "rating_scale_min": 0,
        "rating_scale_max": 5
    }
  }], 
  ...
}
```

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+import abc
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py]: plz add:
``` 
from __future__ import absolute_import, print_function, unicode_literals, division
```

**Comment by kinnou02** [comment]
And there is some conflict now :(

**Comment by xlqian** [comment]
Jenkins, test this please!

File: source/jormungandr/jormungandr/scenarios/ridesharing/tests/instant_system_tests.py [hunk 2]
```diff
@@ -168,15 +168,52 @@
...
+    assert services[1].service_url == 'tata'
+    assert services[1].api_key == 'tata key'
+    assert services[1].network == 'M'
+
+
 def instant_system_test():
     with mock.patch('requests.get', mock_get):
 
-        instant_system = InstantSystem('dummyUrl', 'dummyApiKey')
+        instant_system = InstantSystem('dummyInstance', 'dummyUrl', 'dummyApiKey', 'dummyNetwork')
```
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/ridesharing/tests/instant_system_tests.py]: for later understanding : use class DummyInstance

**Review by pbougue — APPROVED** [review]

**Review by kinnou02 — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py [hunk 3]
```diff
@@ -121,11 +132,15 @@ def _make_response(self, raw_json):
...
-                                            gender=user.get('gender'),
-                                            image_url=user.get('imageUrl'),
-                                            rate=user.get('rate'),
-                                            rate_count=user.get('count'))
+                gender = user.get('gender')
+                gender_map = {'MALE': rsj.Gender.MALE,
+                              'FEMALE': rsj.Gender.FEMALE}
+
+                res.driver = rsj.Individual(alias=user.get('alias'),
+                                            gender=gender_map.get(gender, rsj.Gender.UNKNOWN),
```
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py:140]: I guess None would do just the job, no?

**Comment by pbougue** [comment]
also, do we have timeout and downtime management?

File: source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py (hunk 3 above)
Comment (xlqian) [reply source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py]: I hesitated.. i think the code will be more semantic given that enum...

**Comment by xlqian** [comment]
@pbougue if the outside service timeout, an exception will be raise

File: source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py (hunk 3 above)
Comment (xlqian) [reply source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py:140]: Yes...

**Pull request merged**